pkg/cluster/hive_test.go

```diff
@@ -272,3 +272,52 @@ func createManagerForTests(t *testing.T, existingNamespaceName string) *manager
 	}
 	return m
 }
+
+func TestHiveEnsureResources(t *testing.T) {
+	ctx := context.Background()
+	for _, tt := range []struct {
+		testName           string
+		clusterManagerMock func(mockCtrl *gomock.Controller, m *manager) *mock_hive.MockClusterManager
+		wantErr            string
+	}{
+		{
+			testName: "returns error if cluster manager returns error",
+			clusterManagerMock: func(mockCtrl *gomock.Controller, m *manager) *mock_hive.MockClusterManager {
+				mockClusterManager := mock_hive.NewMockClusterManager(mockCtrl)
+				mockClusterManager.EXPECT().CreateOrUpdate(gomock.Any(), m.subscriptionDoc, m.doc).Return(fmt.Errorf("cluster manager error"))
```

Comment:
Nit: let's either use `gomock.Any()` for `ctx` arg or assert it in both places.

```suggestion
				mockClusterManager.EXPECT().CreateOrUpdate(ctx, m.subscriptionDoc, m.doc).Return(fmt.Errorf("cluster manager error"))
```

Reply:
I'm not sure I agree. The purpose of this unit test is not to assert parameters. This should be `gomock.Any()` for all params, which I will fix in a sec.